feature/settings/src/main/java/com/google/jetpackcamera/settings/SettingsViewModel.kt

```diff
@@ -51,14 +59,279 @@ class SettingsViewModel @Inject constructor(
         ) { updatedSettings, constraints ->
             SettingsUiState.Enabled(
                 cameraAppSettings = updatedSettings,
-                systemConstraints = constraints
+                systemConstraints = constraints,
```

Comment:
If we thoroughly define our `SettingsUiState`, we should no longer need the `CameraAppSettings` or `SystemConstraints` in `SettingsUiState`. The entire state of every setting will be calculated in the `ViewModel`, and the `SettingsUiState` will define exactly what should be shown in the UI.

Reply:
yup, i'm working on just that! It'll be in the updated the PR when I finish addressing your other comments and some other cleanup.